packages/lit-html/src/directive-helpers.ts

```diff
@@ -118,12 +118,19 @@ export const insertPart = (
     const endNode = wrap(part._$endNode!).nextSibling;
     const parentChanged = part._$parent !== containerPart;
     if (parentChanged) {
+      const newConnectionState = containerPart._$isConnected;
+      const shouldNotifyConnectionChanged =
+        part._$notifyConnectionChanged !== undefined &&
+        newConnectionState !== part._$isConnected;
       part._$reparentDisconnectables?.(containerPart);
       // Note that although `_$reparentDisconnectables` updates the part's
       // `_$parent` reference after unlinking from its current parent, that
       // method only exists if Disconnectables are present, so we need to
       // unconditionally set it here
       part._$parent = containerPart;
+      if (shouldNotifyConnectionChanged) {
+        part._$notifyConnectionChanged?.(newConnectionState);
```

Comment:
You don't need both the optional chaining here and the `part._$notifyConnectionChanged !== undefined` above.

Reply:
Removed, was dead code.